titan-compiler/checker.lua

```diff
@@ -137,16 +200,15 @@ check_type = function(typ, errors)
     elseif tag == ast.Type.Name then
         local decl = typ._decl
         if decl._tag == ast.Toplevel.Record then
-            return decl._type
+            return assert(decl._type)
         else
-            type_error(errors, typ.loc, "'%s' isn't a type", typ.name)
-            return types.T.Invalid()
+            type_error(typ.loc, "'%s' isn't a type", typ.name)
```

Comment:
Couldn't we have a type_assert function? Most of the cases now are something like:
```
if something wrong then
  type_error(...)
end
```

Reply:
Some cases don't work with type_assert. For example, sometimes the type error appears as part of an ifelseif chain:

```
if bla._tag == "AAA" then
   type error
elseif bla._tag == "BBB" then
   do something
else
   error("impossible")
end
```

Reply:
We can have both, not only one.